---
title: "2025.5.28-c: Azenta Amplicon-EZ (part-3)"
output: html_document
---


## Samples

<img src="https://raw.githubusercontent.com/chenh19/chemProbing/refs/heads/main/images/2025.5.28/Samples.png" width="100%">
  
RNA covalent modifiers:  

<img src="https://raw.githubusercontent.com/chenh19/chemProbing/refs/heads/main/images/2025.5.28/RNA_CM.png" width="50%">

- The **Amplicon-EZ** service was provided by [Azenta](https://www.genewiz.com/public/services/next-generation-sequencing/amplicon-sequencing-services/amplicon-ez).  

- For experimental details, click [here](https://raw.githubusercontent.com/chenh19/chemProbing/refs/heads/main/files/2025.5.28/2025.5.12 80C samples re-seq with Azenta Amplicon-EZ.pdf).


## Reference sequences

- [Amplicon-3-trimmed.fa](https://raw.githubusercontent.com/chenh19/chemProbing/refs/heads/main/files/2025.5.28/3.SMN2/Amplicon-2.fa)  
```{r,eval=FALSE}
>SMN2-Exon7-Amplicon  (324 bp)
AACTTCCTTTATTTTCCTTACAGGGTTTTAGACAAAATCAAAAAGAAGGAAGGTGCTCACATTCCTTAAATTAAGGAGTA
AGTCTGCCAGCATTATGAAAGTGAATCTTACTTTTGTAAAACTTTATGGTTTGTGGAAAACAAATGTTTTTGAACATTTA
AAAAGTTCAGATGTTAGAAAGTTGAAAGGTTAATGTAAAACAATCAATATTAAAGAATTTTGATGCCAAAACTATTAGAT
AAAAGGTTAATCTACATCCCTACTAGAATTCTCATACTTAACTGGTTGGTTGTGTGGAAGAAACATACTTTCACAATAAA
GAGC
```

- **Prrepare reference sequences with ** [```1.refseq.sh```](https://raw.githubusercontent.com/chenh19/chemProbing/refs/heads/main/files/2025.5.28/3.SMN2/1.refseq.sh)


## FastQC for reads

- **FastQC with ** [```2.fastqc.sh```](https://raw.githubusercontent.com/chenh19/chemProbing/refs/heads/main/files/2025.5.28/3.SMN2/2.fastqc.sh)  

- Sample-13_R1 (representative FastQC per base quality plot before reads trimming):  
<img src="https://raw.githubusercontent.com/chenh19/chemProbing/refs/heads/main/images/2025.5.28/3.SMN2/1.fastqc/13-Ctrl-DMSO-80C-5min-Mn_R1_001_fastqc_per_base_quality.png" width="80%">
- Sample-13_R2 (representative FastQC per base quality plot before reads trimming):  
<img src="https://raw.githubusercontent.com/chenh19/chemProbing/refs/heads/main/images/2025.5.28/3.SMN2/1.fastqc/13-Ctrl-DMSO-80C-5min-Mn_R2_001_fastqc_per_base_quality.png" width="80%">

<details>
<summary>View all samples before reads trimming:</summary>

- Sample-14_R1:  
<img src="https://raw.githubusercontent.com/chenh19/chemProbing/refs/heads/main/images/2025.5.28/3.SMN2/1.fastqc/14-Ctrl-C12-100uM-80C-5min-Mn_R1_001_fastqc_per_base_quality.png" width="80%">
- Sample-14_R2:  
<img src="https://raw.githubusercontent.com/chenh19/chemProbing/refs/heads/main/images/2025.5.28/3.SMN2/1.fastqc/14-Ctrl-C12-100uM-80C-5min-Mn_R2_001_fastqc_per_base_quality.png" width="80%">

- Sample-15_R1:  
<img src="https://raw.githubusercontent.com/chenh19/chemProbing/refs/heads/main/images/2025.5.28/3.SMN2/1.fastqc/15-Ctrl-C15-100uM-80C-5min-Mn_R1_001_fastqc_per_base_quality.png" width="80%">
- Sample-15_R2:  
<img src="https://raw.githubusercontent.com/chenh19/chemProbing/refs/heads/main/images/2025.5.28/3.SMN2/1.fastqc/15-Ctrl-C15-100uM-80C-5min-Mn_R2_001_fastqc_per_base_quality.png" width="80%">

</details>

- **Conclusion**: within 250bp, the reads quality is generally good. R2 has slightly lower overall quality than R1,which is expected for paired-end reads.


## Reads trimming

- **Reads trimming with ** [```3.trim.sh```](https://raw.githubusercontent.com/chenh19/chemProbing/refs/heads/main/files/2025.5.28/3.SMN2/3.trim.sh)  

- Key filtering standards:  
```{r,eval=FALSE}
1.Keeping the first 200bp and discard the tail
2.Trimming the tail with average Q-score < 20 (4 bases moving window size)
3.Discarding reads with more than 20% of Q-score < 20 bases
```


## FastQC for reads after trimming

- **Re-FastQC with ** [```4.refastqc.sh```](https://raw.githubusercontent.com/chenh19/chemProbing/refs/heads/main/files/2025.5.28/3.SMN2/4.refastqc.sh)  

- Sample-7_R1 (representative FastQC per base quality plot after reads trimming):  
<img src="https://raw.githubusercontent.com/chenh19/chemProbing/refs/heads/main/images/2025.5.28/3.SMN2/2.refastqc/13-Ctrl-DMSO-80C-5min-Mn_001_R1.trimmed_fastqc_per_base_quality.png" width="80%">
- Sample-7_R2 (representative FastQC per base quality plot after reads trimming):  
<img src="https://raw.githubusercontent.com/chenh19/chemProbing/refs/heads/main/images/2025.5.28/3.SMN2/2.refastqc/13-Ctrl-DMSO-80C-5min-Mn_001_R2.trimmed_fastqc_per_base_quality.png" width="80%">

<details>
<summary>View all samples after reads trimming:</summary>

- Sample-8_R1:  
<img src="https://raw.githubusercontent.com/chenh19/chemProbing/refs/heads/main/images/2025.5.28/3.SMN2/2.refastqc/14-Ctrl-C12-100uM-80C-5min-Mn_001_R1.trimmed_fastqc_per_base_quality.png" width="80%">
- Sample-8_R2:  
<img src="https://raw.githubusercontent.com/chenh19/chemProbing/refs/heads/main/images/2025.5.28/3.SMN2/2.refastqc/14-Ctrl-C12-100uM-80C-5min-Mn_001_R2.trimmed_fastqc_per_base_quality.png" width="80%">

- Sample-9_R1:  
<img src="https://raw.githubusercontent.com/chenh19/chemProbing/refs/heads/main/images/2025.5.28/3.SMN2/2.refastqc/15-Ctrl-C15-100uM-80C-5min-Mn_001_R1.trimmed_fastqc_per_base_quality.png" width="80%">
- Sample-9_R2:  
<img src="https://raw.githubusercontent.com/chenh19/chemProbing/refs/heads/main/images/2025.5.28/3.SMN2/2.refastqc/15-Ctrl-C15-100uM-80C-5min-Mn_001_R2.trimmed_fastqc_per_base_quality.png" width="80%">

</details>

- **Conclusion**: after trimming, the quality of the overall reads quality is improved and variation is reduced.


## Reads mapping

- **Reads mapping with ** [```5.bam.sh```](https://raw.githubusercontent.com/chenh19/chemProbing/refs/heads/main/files/2025.5.28/3.SMN2/5.bam.sh)  


## Analyzing and visualization

- **Pileup with ** [```6.mpileup.sh```](https://raw.githubusercontent.com/chenh19/chemProbing/refs/heads/main/files/2025.5.28/3.SMN2/6.mpileup.sh)  
- **Parse with ** [```7.parse.R```](https://raw.githubusercontent.com/chenh19/chemProbing/refs/heads/main/files/2025.5.28/3.SMN2/7.parse.R)  
- For parsed mutation rate spreadsheets, click [here](https://raw.githubusercontent.com/chenh19/chemProbing/refs/heads/main/files/2025.5.28/3.SMN2/mpileup_parse.zip).
- **Plot with ** [```8.plot.R```](https://raw.githubusercontent.com/chenh19/chemProbing/refs/heads/main/files/2025.5.28/3.SMN2/8.plot.R)  

### Absolute mutation rate (after reads trimming)

<details>
<summary>Absolute mutation rate:</summary>

- Sample-13:  
<img src="https://raw.githubusercontent.com/chenh19/chemProbing/refs/heads/main/images/2025.5.28/3.SMN2/3.absolute_mut/13-Ctrl-DMSO-80C-5min-Mn-001_SMN2-Exon7-Amplicon.jpg" width="100%">

- Sample-14:  
<img src="https://raw.githubusercontent.com/chenh19/chemProbing/refs/heads/main/images/2025.5.28/3.SMN2/3.absolute_mut/14-Ctrl-C12-100uM-80C-5min-Mn-001_SMN2-Exon7-Amplicon.jpg width="100%">

- Sample-15:  
<img src="https://raw.githubusercontent.com/chenh19/chemProbing/refs/heads/main/images/2025.5.28/3.SMN2/3.absolute_mut/15-Ctrl-C15-100uM-80C-5min-Mn-001_SMN2-Exon7-Amplicon.jpg" width="100%">

</details>

### Relative mutation rate (after reads trimming)

<details>
<summary>Relative mutation rate:</summary>

- Sample-13:  
<img src="https://raw.githubusercontent.com/chenh19/chemProbing/refs/heads/main/images/2025.5.28/3.SMN2/4.relative_mut/13-Ctrl-DMSO-80C-5min-Mn-001_SMN2-Exon7-Amplicon.jpg" width="100%">

- Sample-14:  
<img src="https://raw.githubusercontent.com/chenh19/chemProbing/refs/heads/main/images/2025.5.28/3.SMN2/4.relative_mut/14-Ctrl-C12-100uM-80C-5min-Mn-001_SMN2-Exon7-Amplicon.jpg" width="100%">

- Sample-15:  
<img src="https://raw.githubusercontent.com/chenh19/chemProbing/refs/heads/main/images/2025.5.28/3.SMN2/4.relative_mut/15-Ctrl-C15-100uM-80C-5min-Mn-001_SMN2-Exon7-Amplicon.jpg" width="100%">

</details>

### Base Summary

- All samples:  

<img src="https://raw.githubusercontent.com/chenh19/chemProbing/refs/heads/main/images/2025.5.28/3.SMN2/5.base_summary/base_summary_ctrl.jpg" width="60%">

- **Conclusion**: Overall result is consistent with [previous experiment](https://chenh19.github.io/chemProbing/2025.5.27-c.html#base-summary). 
