---
title: "2025.5.27-c: Plasmidsaurus data re-analysis (part-3)"
output: html_document
---


## Samples

- Re-analysis of [2025.5.7 dataset](https://chenh19.github.io/chemProbing/2025.5.7.html) with reads trimming.
- Controls using Zhichao's cDNA.

## Reference sequences

- [Amplicon-2.fa](https://raw.githubusercontent.com/chenh19/chemProbing/refs/heads/main/files/2025.5.27/3.SMN2/Amplicon-2.fa)  
```{r,eval=FALSE}
>SMN2-Exon7-Amplicon  (324 bp)
AACTTCCTTTATTTTCCTTACAGGGTTTTAGACAAAATCAAAAAGAAGGAAGGTGCTCACATTCCTTAAATTAAGGAGTA
AGTCTGCCAGCATTATGAAAGTGAATCTTACTTTTGTAAAACTTTATGGTTTGTGGAAAACAAATGTTTTTGAACATTTA
AAAAGTTCAGATGTTAGAAAGTTGAAAGGTTAATGTAAAACAATCAATATTAAAGAATTTTGATGCCAAAACTATTAGAT
AAAAGGTTAATCTACATCCCTACTAGAATTCTCATACTTAACTGGTTGGTTGTGTGGAAGAAACATACTTTCACAATAAA
GAGC
```

- **Prrepare reference sequences with ** [```1.refseq.sh```](https://raw.githubusercontent.com/chenh19/chemProbing/refs/heads/main/files/2025.5.27/3.SMN2/1.refseq.sh)


## FastQC for reads

- **FastQC with ** [```2.fastqc.sh```](https://raw.githubusercontent.com/chenh19/chemProbing/refs/heads/main/files/2025.5.27/3.SMN2/2.fastqc.sh)  

- Sample-13 (representative FastQC per base quality plot before reads trimming):  
<img src="https://raw.githubusercontent.com/chenh19/chemProbing/refs/heads/main/images/2025.5.27/3.SMN2/1.fastqc/WV2JYX_13_13_Ctrl_DMSO_80C_5min_Mn_fastqc_per_base_quality.png" width="80%">


<details>
<summary>View all samples before reads trimming:</summary>

- Sample-14:  
<img src="https://raw.githubusercontent.com/chenh19/chemProbing/refs/heads/main/images/2025.5.27/3.SMN2/1.fastqc/WV2JYX_14_14_Ctrl_C12_0.1mM_80C_5min_Mn_fastqc_per_base_quality.png" width="80%">

- Sample-15:  
<img src="https://raw.githubusercontent.com/chenh19/chemProbing/refs/heads/main/images/2025.5.27/3.SMN2/1.fastqc/WV2JYX_15_15_Ctrl_C15_0.1mM_80C_5min_Mn_fastqc_per_base_quality.png" width="80%">

</details>

- **Conclusion**: within ~400bp, the reads quality is generally good, but the variation is big.


## Reads trimming

- **Reads trimming with ** [```3.trim.sh```](https://raw.githubusercontent.com/chenh19/chemProbing/refs/heads/main/files/2025.5.27/3.SMN2/3.trim.sh)  

- Key filtering standards:  
```{r,eval=FALSE}
1.Keeping the first 400bp and discard the tail
2.Trimming the tail with average Q-score < 20 (4 bases moving window size)
3.Discarding reads with more than 20% of Q-score < 20 bases
4.Discarding reads less than 200bp
```


## FastQC for reads after trimming

- **Re-FastQC with ** [```4.refastqc.sh```](https://raw.githubusercontent.com/chenh19/chemProbing/refs/heads/main/files/2025.5.27/3.SMN2/4.refastqc.sh)  

- Sample-13 (representative FastQC per base quality plot after reads trimming): 
<img src="https://raw.githubusercontent.com/chenh19/chemProbing/refs/heads/main/images/2025.5.27/3.SMN2/2.refastqc/WV2JYX_13_13_Ctrl_DMSO_80C_5min_Mn_cut_trimmed_fastqc_per_base_quality.png" width="80%">


<details>
<summary>View all samples before after trimming:</summary>

- Sample-14:  
<img src="https://raw.githubusercontent.com/chenh19/chemProbing/refs/heads/main/images/2025.5.27/3.SMN2/2.refastqc/WV2JYX_14_14_Ctrl_C12_0.1mM_80C_5min_Mn_cut_trimmed_fastqc_per_base_quality.png" width="80%">

- Sample-15:  
<img src="https://raw.githubusercontent.com/chenh19/chemProbing/refs/heads/main/images/2025.5.27/3.SMN2/2.refastqc/WV2JYX_15_15_Ctrl_C15_0.1mM_80C_5min_Mn_cut_trimmed_fastqc_per_base_quality.png" width="80%">

</details>

- **Conclusion**: after trimming, the quality of the overall reads quality is improved and variation is reduced.


## Reads mapping

- **Reads mapping with ** [```5.bam.sh```](https://raw.githubusercontent.com/chenh19/chemProbing/refs/heads/main/files/2025.5.27/3.SMN2/5.bam.sh)  
- **Note**: [Minimap2](https://doi.org/10.1093/bioinformatics/bty191) was the aligning tool used here ([GitHub](https://github.com/lh3/minimap2)).It's fast and suited for Nanopore reads (doesn't require R1 and R2 as input).  


## Analyzing and visualization

- **Pileup with ** [```6.mpileup.sh```](https://raw.githubusercontent.com/chenh19/chemProbing/refs/heads/main/files/2025.5.27/3.SMN2/6.mpileup.sh)  
- **Parse with ** [```7.parse.R```](https://raw.githubusercontent.com/chenh19/chemProbing/refs/heads/main/files/2025.5.27/3.SMN2/7.parse.R)  
- For parsed mutation rate spreadsheets, click [here](https://raw.githubusercontent.com/chenh19/chemProbing/refs/heads/main/files/2025.5.27/3.SMN2/pileup_parse.zip).
- **Plot with ** [```8.plot.R```](https://raw.githubusercontent.com/chenh19/chemProbing/refs/heads/main/files/2025.5.27/3.SMN2/8.plot.R)  

### Absolute mutation rate (after reads trimming)

<details>
<summary>Absolute mutation rate:</summary>

- Sample-13:  
<img src="https://raw.githubusercontent.com/chenh19/chemProbing/refs/heads/main/images/2025.5.27/3.SMN2/3.absolute_mut/13-13-Ctrl-DMSO-80C-5min-Mn_SMN2-Exon7-Amplicon.jpg" width="100%">

- Sample-14:  
<img src="https://raw.githubusercontent.com/chenh19/chemProbing/refs/heads/main/images/2025.5.27/3.SMN2/3.absolute_mut/14-14-Ctrl-C12-0.1mM-80C-5min-Mn_SMN2-Exon7-Amplicon.jpg" width="100%">

- Sample-15:  
<img src="https://raw.githubusercontent.com/chenh19/chemProbing/refs/heads/main/images/2025.5.27/3.SMN2/3.absolute_mut/15-15-Ctrl-C15-0.1mM-80C-5min-Mn_SMN2-Exon7-Amplicon.jpg" width="100%">

</details>

### Relative mutation rate (after reads trimming)

<details>
<summary>Relative mutation rate:</summary>

- Sample-13:  
<img src="https://raw.githubusercontent.com/chenh19/chemProbing/refs/heads/main/images/2025.5.27/3.SMN2/4.relative_mut/13-13-Ctrl-DMSO-80C-5min-Mn_SMN2-Exon7-Amplicon.jpg" width="100%">

- Sample-14:  
<img src="https://raw.githubusercontent.com/chenh19/chemProbing/refs/heads/main/images/2025.5.27/3.SMN2/4.relative_mut/14-14-Ctrl-C12-0.1mM-80C-5min-Mn_SMN2-Exon7-Amplicon.jpg" width="100%">

- Sample-15:  
<img src="https://raw.githubusercontent.com/chenh19/chemProbing/refs/heads/main/images/2025.5.27/3.SMN2/4.relative_mut/15-15-Ctrl-C15-0.1mM-80C-5min-Mn_SMN2-Exon7-Amplicon.jpg" width="100%">

</details>

### Base Summary

- All samples:  

<img src="https://raw.githubusercontent.com/chenh19/chemProbing/refs/heads/main/images/2025.5.27/3.SMN2/5.base_summary/base_summary.jpg" width="50%">

- **Conclusion**: The trend of mutations is consistent, but maybe there is a degradation in the old samples.
